homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -19,14 +19,21 @@ class YaleDataUpdateCoordinator(DataUpdateCoordinator):
     def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
         """Initialize the Yale hub."""
         self.entry = entry
-        self.yale: YaleSmartAlarmClient | None = None
+        self.yale: YaleSmartAlarmClient = None
```

Comment:
Please revert or rework. The typing isn't complete with that change.

Reply:
I did not get it before but now when I have the original I get the following mypy errors which I have no clue how to fix:
homeassistant/components/yale_smart_alarm/coordinator.py:139: error: Item "None" of "Optional[Any]" has no attribute "get_armed_status"  [union-attr]
homeassistant/components/yale_smart_alarm/coordinator.py:140: error: Item "None" of "Optional[Any]" has no attribute "get_cycle"  [union-attr]
homeassistant/components/yale_smart_alarm/coordinator.py:141: error: Item "None" of "Optional[Any]" has no attribute "get_status"  [union-attr]
homeassistant/components/yale_smart_alarm/coordinator.py:142: error: Item "None" of "Optional[Any]" has no attribute "get_online"  [union-attr]

Reply:
We need to check or assert the type before we use the instance attribute since it's optional.

Reply:
Or create the client before creating the coordinator and pass the client to the coordinator.

Reply:
Thanks for explaning. Made sense now when I understood.